File: programmers/greedy/2차원-동전-뒤집기.py
def solution(beginning, target):
    m, n = len(beginning), len(beginning[0])
    count = 0

    # row flips 결정: 첫 번째 열을 맞추는 방식
    for x in range(m):
        if beginning[x][0] != target[x][0]:
            count += 1
            for y in range(n):
                beginning[x][y] ^= 1

    # col flips 결정: 첫 번째 행을 맞추는 방식
    for y in range(n):
        if beginning[0][y] != target[0][y]:
            count += 1
            for x in range(m):
                beginning[x][y] ^= 1

    # 최종 검증 (여기서만)
    for x in range(m):
        for y in range(n):
            if beginning[x][y] != target[x][y]:
                return -1

    # (R, C)와 (complement R, complement C) 둘 다 가능하므로 최소 선택
    return min(count, m + n - count)

File: programmers/greedy/2차원-동전-뒤집기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> beginning, vector<vector<int>> target) {
    
    int m = target.size(), n = target[0].size();
    int flip = 0;
    
    for (int i = 0; i < m; i++) {
        if (beginning[i][0] == target[i][0]) 
            continue;
        
        for (int j = 0; j < n; j++) {
            beginning[i][j] ^= 1;
        }
        flip++;
    }
    
    for (int j = 0; j < n; j++) {
        if (beginning[0][j] == target[0][j]) 
            continue;
        for (int i = 0; i < m; i++) {
            beginning[i][j] ^= 1;
        }
        flip++;
    }
    
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            if (beginning[i][j] != target[i][j])
                return -1;
        }
    }

    return min(m + n - flip, flip);
}
